eds/plugins/github_provider.py

```diff
@@ -0,0 +1,143 @@
+import os
+from typing import List, Dict, Optional
+
+from github3 import login, enterprise_login
+from github3.github import GitHub, GitHubEnterprise
+from github3.orgs import Organization
+from github3.repos.repo import Repository
+from github3.repos.contents import Contents
+# from github3.repos.branch import Branch
+# from github3.git import CommitTree, Commit, Tree, Hash, Reference
+# from github3.repos.comparison import Comparison
+# from github3.users import User
+# from github3.exceptions import NotFoundError, ForbiddenError
+
+from eds.interfaces.vcs_provider import VcsProvider
+from eds.interfaces.plugin import Plugin
+
+
+class GithubProvider(VcsProvider):
+    """Github Provider implementation."""
+
+    def __init__(self, token_env_var: str = None, github_enterprise_url: str = None):
+        """Login to Github or Github Enterprise."""
+        token: str = os.environ.get(token_env_var)
+        if token == '' or token is None:
+            raise RuntimeError(
+                f'ERROR: You must export the {token_env_var} environment variable.'
+            )
+        if github_enterprise_url is None:
+            self._g: GitHub = login(token=token)
+        else:
+            self._g: GitHubEnterprise = enterprise_login(url=github_enterprise_url, token=token)
+
+    @property
+    def children(self) -> List[Plugin]:
+        """The list of child plugins.
+
+        Returns:
+            List[Plugin]: The list of child plugins.
+        """
+        return []
+
+    def validate(self) -> None:
+        """Validate the plugin."""
+        return super().validate()
+
+    def parse_event(self) -> Dict:
+        """Parse webhook event for project url and ref."""
+        return super().parse_event()
```

Comment:
there's no parent `parse_event` to call.   the intention of this method is to parse project url and sha from the webhook payload.

https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads

Reply:
I implemented a very simple version of `parse_event` for now. I think we can continue to update the functionality as we work on more issues.